src/t_erfa_c_extra.c

```diff
@@ -0,0 +1,167 @@
+/*
+** Copyright (C) 2019, NumFOCUS Foundation.
+**
+** Licensed under a 3-clause BSD style license - see LICENSE
+**
+** This file is NOT derived from SOFA sources.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <erfa.h>
+#include <erfaextra.h>
+
+static int verbose = 0;
+
+/*
+**
+**  Validate the ERFA C functions that are not derived from SOFA (SOFA-derived tests are in t_erfa_c)
+**
+*/
+
+static void t_versions(int *status)
+/*
+**  Test that the version-checking functions yield something.
+*/
+{
+  char buf[3];
+  char* res;
+  const char* version_str = eraVersion();
+
+  sprintf(buf, "%d", eraVersionMajor());
+  res = strstr(version_str, buf);
+  if (!res) {
+    *status = 1;
+    printf("t_versions failed - major version %d not in version string %s\n", res, version_str);
+  }
+
+  sprintf(buf, "%d", eraVersionMinor());
+  res = strstr(version_str, buf);
+  if (!res) {
+    *status = 1;
+    printf("t_versions failed - minor version %d not in version string %s\n", res, version_str);
+  }
+
+  sprintf(buf, "%d", eraVersionMicro());
+  res = strstr(version_str, buf);
+  if (!res) {
+    *status = 1;
+    printf("t_versions failed - micro version %d not in version string %s\n", res, version_str);
+  }
+
+  if (*status == 0) {
+    printf("t_versions passed");
+  }
+
+}
+
+static void t_leap_seconds(int *status)
+/*
+**  Test that the leap-second machinery yields something
+*/
+{
+  int count_init, count_postset;
+  eraLEAPSECOND* leapseconds_init;
+  eraLEAPSECOND* leapseconds_postset;
+  eraLEAPSECOND new_leapsecond[1] = {{ 2050, 5, 35. }};
+
+  count_init = eraGetLeapSeconds(&leapseconds_init);
+  eraSetLeapSeconds(&new_leapsecond, 1);
```

Comment:
@eteq - `eraSetLeapSeconds` replaces the whole leap second table, so the expected number of leap seconds here is 1.

Reply:
Aha, that was the trouble! (that and an incorrect variable)